src/macrofier/engine.mjs

```diff
@@ -402,6 +406,7 @@ const insertMacros = (fContents = '', macros = {}) => {
   fContents = fContents.replace(/[ \t]*\/\* \$\{TYPES\} \*\/[ \t]*\n/, macros.types)
   fContents = fContents.replace(/[ \t]*\/\* \$\{PROVIDERS\} \*\/[ \t]*\n/, macros.providerInterfaces)
   fContents = fContents.replace(/[ \t]*\/\* \$\{ENUMS\} \*\/[ \t]*\n/, macros.enums)
+  fContents = fContents.replace(/[ \t]*\/\* \$\{ENUMS_CONVERSION\} \*\/[ \t]*\n/, macros.enumsConversion)
```

Comment:
So this purpose of this macro is to generate code that enables converting an `enum` from C into the JSON library that we are leveraging in C?

Is there a way to generate this conversion based on the `destination` context? We know which template file we are inserting the schemas into based on `options.destination`, so if we can detect it, instead of creating another macro, that would be ideal.

Reply:
This is to generate enum value string as like below,
ENUM_CONVERSION_BEGIN(SecondScreen_type)
        { SECONDSCREEN_TYPE_DIAL, _T("dial") }
ENUM_CONVERSION_END(SecondScreen_type)

This is cpp specific part and it has to be under WPEFramework namespace, since it is generated based on thunder json module format. https://github.com/rdkcentral/firebolt-openrpc/pull/93/files#diff-97dd97692f7573b953d3489feab237c014a7b0719d301ce1fbd1ec15f9f2e368R33

Hence added a specific macro ENUMS_MACRO

Reply:
So the Module.cpp template does not have any instance of the existing ${ENUMS} macro.

Therefor, we can use the `${ENUMS}` macro and check the value of `options.destination`. If it's `Module.cpp` then we insert the conversion with the code in this PR. 

That way we do not introduce a new macro just to fulfill a particular implementation pattern that we've chosen.

Can we drop the `${ENUMS_CONVERSION}` macro and add logic to do an `if/else` on `options.destination`?

Reply:
yes, I have removed ENUMS_CONVERSION and used ENUMS itself based on destination values